packages/app/src/cli/services/context.ts

```diff
@@ -288,9 +290,68 @@ export async function ensureThemeExtensionDevContext(
   return registration
 }
 
+export type DeploymentMode = 'legacy' | 'unified' | 'unified-skip-release'
+
 export async function ensureDeployContext(options: DeployContextOptions): Promise<DeployContextOutput> {
   const token = await ensureAuthenticatedPartners()
   const [partnersApp, envIdentifiers] = await fetchAppAndIdentifiers(options, token)
+  const noRelease = options.noRelease
+  const deploymentMode: DeploymentMode = (function () {
+    if (partnersApp.betas?.unifiedAppDeployment) {
+      if (noRelease) {
+        return 'unified-skip-release'
+      } else {
+        return 'unified'
+      }
+    } else {
+      return 'legacy'
+    }
+  })()
+
+  if (deploymentMode === 'legacy') {
```

Comment:
Good points!
> Also, if we ever remove the unifiedAppDeployment beta flag, this code would make the client go back to the 'legacy' version as undefined is falsey.

Could the function that calculates the `deploymentMode` value be changed to assume that an `undefined` value means that the beta flag has been removed (to prevent `deploymentMode` from falling back to the `legacy` value)? I.e.:
```
  const deploymentMode: DeploymentMode = (function () {
    if (partnersApp.betas?.unifiedAppDeployment || (typeof partnersApp.betas?.unifiedAppDeployment === "undefined")) {
      if (noRelease) {
        return 'unified-skip-release'
      } else {
        return 'unified'
      }
    } else {
      return 'legacy'
    }
  })()
```

> I heard that we'd prefer to always show a message. If so, wouldn't it be a bit redundant to show it to someone that has opted in and has know about the feature for a while?

I'll defer to @jessehoek /others, but how do other versioned packages like the CLI deal with temporary info messages?